crates/iota-indexer/src/db.rs

```diff
@@ -311,11 +363,86 @@ pub mod setup_postgres {
             let store = PgIndexerStore::new(blocking_cp, indexer_metrics.clone());
             return Indexer::start_writer(&indexer_config, store, indexer_metrics).await;
         } else if indexer_config.rpc_server_worker {
+            check_db_migration_consistency(&mut conn)?;
             return Indexer::start_reader(&indexer_config, &registry, db_url.to_string()).await;
         } else if indexer_config.analytical_worker {
             let store = PgIndexerAnalyticalStore::new(blocking_cp);
             return Indexer::start_analytical_worker(store, indexer_metrics.clone()).await;
         }
         Ok(())
     }
+
+    #[cfg(feature = "pg_integration")]
+    #[cfg(test)]
+    mod tests {
+        use diesel::{
+            migration::{Migration, MigrationSource},
+            pg::Pg,
+        };
+        use diesel_migrations::MigrationHarness;
+        use secrecy::Secret;
+
+        use crate::{
+            db::setup_postgres::{self, MIGRATIONS},
+            test_utils::TestDatabase,
+        };
+
+        fn database_url(db_name: &str) -> Secret<String> {
+            secrecy::Secret::new(format!(
+                "postgres://postgres:postgrespw@localhost:5432/{db_name}"
+            ))
+        }
+
+        // Check that the migration records in the database created from the local
+        // schema pass the consistency check.
+        #[test]
+        fn db_migration_consistency_smoke_test() {
+            let mut database =
+                TestDatabase::new(database_url("db_migration_consistency_smoke_test"));
+            database.recreate();
+            let pool = database.to_connection_pool();
+            let mut conn = pool.get().unwrap();
+            setup_postgres::reset_database(&mut conn).unwrap();
+            setup_postgres::check_db_migration_consistency(&mut conn).unwrap();
```

Comment:
```suggestion
            setup_postgres::check_db_migration_consistency(&mut conn).unwrap();
            database.drop_if_exists();
```

Reply:
Please drop accordingly the test databases in the other tests as well.

Reply:
https://github.com/iotaledger/iota/pull/7146/commits/fbf2ac857869957e304a80399ca6ff14640ee970